pkg/ciliumenvoyconfig/cec_resource_parser.go

```diff
@@ -381,13 +389,18 @@ func (r *cecResourceParser) parseResources(cecNamespace string, cecName string,
 		}
 	}
 
+	injectCiliumUpstreamFilters := false
+
 	// Allocate TPROXY ports for listeners without address.
 	// Do this only after all other possible error cases.
 	for _, listener := range resources.Listeners {
 		// Figure out if this is an internal listener
 		isInternalListener := listener.GetInternalListener() != nil
 
 		if !isInternalListener {
+			// upstream filters are injected if any non-internal listener is L7 LB
+			injectCiliumUpstreamFilters = injectCiliumUpstreamFilters || isL7LB
```

Comment:
do we have to add the condition whether any downstream cilium filter has been added to the listener? (e.g. edge case if listener address isn't set by Cilium)

hence only add upstream filter if a downstream filter is present? or doesn't this matter?